src/components/renderer/file-download.vue

```diff
@@ -9,97 +9,108 @@
     </b-card>
     <div v-else>
       <template v-if="filesInfo.length > 0">
-        <div v-for="(file, idx) in filesInfo" :key="idx" :data-cy="file.id + '-' + file.name.replace(/[^0-9a-zA-Z\-]/g, '-')">
-          <b-btn v-show="!isReadOnly"
-            class="mb-2 d-print-none" variant="primary" :aria-label="$attrs['aria-label']"
-            @click="downloadFile(file)"
-          >
-            <i class="fas fa-file-download"/> {{ $t('Download') }}
+        <div v-for="(file, idx) in filesInfo" :key="idx"
+          :data-cy="file.id + '-' + file.name.replace(/[^0-9a-zA-Z\-]/g, '-')">
+          <b-btn v-show="!isReadOnly" class="mb-2 d-print-none" variant="primary" :aria-label="$attrs['aria-label']"
+            @click="downloadFile(file)">
+            <i class="fas fa-file-download" /> {{ $t("Download") }}
           </b-btn>
           {{ file.file_name }}
         </div>
       </template>
       <div v-else>
-        {{ $t('No files available for download') }}
+        {{ $t("No files available for download") }}
       </div>
     </div>
   </div>
 </template>
 
-
 <script>
-import _ from  'lodash';
+import _ from "lodash";
 
 export default {
   inheritAttrs: false,
+  props: {
+    name: {
+      type: String,
+      default: ""
+    },
+    value: {
+      type: Number,
+      default: null
+    },
+    endpoint: {
+      type: String,
+      default: null
+    },
+    requestFiles: {
+      type: Array,
+      default: null
+    },
+    label: {
+      type: String,
+      default: ""
+    },
+    transientData: {
+      type: Object,
+      default: null
+    }
+  },
+
   data() {
     return {
       filesInfo: [],
-      prefix: '',
+      prefix: "",
       rowId: null,
+      messageForNotAvailable: this.$t("Downloading files is not available."),
+      messageForPreview: this.$t(
+        "Download button for {{fileName}} will appear here.",
+        {
+          fileName: this.name
+        }
+      ),
+      inPreviewMode:
+        this.$root.$children[0].mode === "preview" && !window.exampleScreens,
+      isReadOnly: this.$attrs.readonly ? this.$attrs.readonly : false,
+      fileDataName:
+        this.prefix + this.name + (this.rowId ? `.${this.rowId}` : ""),
+      requestData: {
+        _parent: { ...this.$parent._parent },
```

Comment:
Please do not use the spread operator "..." because it could cause reactivity problems, consider to use `this.getScreenDataReference()`